clean/baton_rouge_pd_cprr.py

```diff
@@ -504,15 +504,12 @@ def split_name_21(df):
             "det": "detective",
         }
     )
-    df.loc[:, "first_name"] = names[1].str.strip()
-    df.loc[:, "last_name"] = names[3].str.strip()
-    df.loc[:, "middle_name"] = (
-        names.loc[:, 2].str.strip().fillna("").map(lambda s: "" if len(s) < 2 else s)
-    )
-    df.loc[:, "middle_initial"] = (
-        names.loc[:, 2].str.strip().fillna("").map(lambda s: "" if len(s) > 2 else s)
+    df.loc[:, "first_name"] = names[1].str.strip().fillna("")
+    df.loc[:, "last_name"] = (
+        names[3].str.strip().fillna("").str.replace(r"\,", " ", regex=True)
     )
-    return df.drop(columns=["name"])
+    df.loc[:, "middle_name"] = names[2]
+    return df[~((df.first_name == "") & (df.last_name == ""))].drop(columns="name")
```

Comment:
I don't think the `.drop(columns="name")` bit does anything.

Reply:
It seems to be correctly dropping the `name` column after splitting said column into `first_name`, `middle_name`, and `last_name`. I'm going to go ahead and merge.

Reply:
@ayyubibrahimi I tried this at local and doesn't look like you can drop column on a subset of rows:
```python
>>> import pandas as pd
>>> df = pd.DataFrame([[1,2],[3,4]], columns=['a','b'])
>>> df
   a  b
0  1  2
1  3  4
>>> df.loc[0]
a    1
b    2
Name: 0, dtype: int64
>>> df.loc[0].drop(columns='a')
a    1
b    2
Name: 0, dtype: int64
>>> df
   a  b
0  1  2
1  3  4
```
But maybe I'm missing something.

Reply:
This is my output: 

1) `drop(columns="name") `is included in the return statement of the `split_name_21 function`
2) after dropping `drop(columns="name")` from the `split_name_21` function in the baton_rouge_pd_cprr.py script

```
>>> from importlib import reload
>>> from clean import baton_rouge_pd_cprr as n
>>> df = n.clean_21()
standardize_from_lookup_table: unmatched sequences:
  set()
>>> df.columns
Index(['received', 'action', 'disposition', 'complainant',
       'investigation_status', 'tracking_number', 'receive_year',      
       'receive_month', 'receive_day', 'occur_year', 'occur_month',    
       'occur_day', 'allegation', 'department_code', 'department_desc',
       'rank_desc', 'first_name', 'last_name', 'middle_name', 'division_desc',
       'agency', 'data_production_year', 'uid', 'allegation_uid'],
      dtype='object')
>>> reload(n)
<module 'clean.baton_rouge_pd_cprr' from '/mnt/c/Users/ayyubi/lleads/processing/clean/baton_rouge_pd_cprr.py'>
>>> df = n.clean_21()
standardize_from_lookup_table: unmatched sequences:
  set()
>>> df.columns
Index(['received', 'action', 'disposition', 'complainant',
       'investigation_status', 'tracking_number', 'receive_year',
       'receive_month', 'receive_day', 'occur_year', 'occur_month',
       'occur_day', 'allegation', 'name', 'department_code', 'department_desc',
       'rank_desc', 'first_name', 'last_name', 'middle_name', 'division_desc',
       'agency', 'data_production_year', 'uid', 'allegation_uid'],
      dtype='object')
```

Thoughts?

Reply:
It seems that the reason is you were not using `.loc`. It's fine then. Sorry for the false alarm :)

Reply:
Should I be using `.loc` here? Happy to make the change.

Reply:
Oh no, I would say the only time when `.loc` must be used is when assigning. Not using `.loc` in this case does make it shorter. It is up to your judgment then.